### ISSUE TITLE
Error al intentar compilar en Haiku

### ISSUE THREAD

**Issue by unspacyar** [pipagerardo/nMSXtiles#8]
Hola! Intente compilar el programa bajo el sistema operativo Haiku, pero obtengo el siguiente error:

`g++ -c -pipe -O2 -Wall -Wextra -fno-exceptions -fno-rtti -fPIC -DVER_NMSXTILES_MAJOR=0 -DVER_NMSXTILES_MINOR=9 -DVER_NMSXTILES_PATCH=6 -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x060000 -DQT_NO_EXCEPTIONS -DQT_NO_DYNAMIC_CAST -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I. -I. -I../build -I../../../../system/develop/headers/x86/QtWidgets -I../../../../system/develop/headers/x86/QtGui -I../../../../system/develop/headers/x86/QtCore -I. -I. -I../../../../system/data/Qt5/mkspecs/haiku-g++ -o ../obj/mainwindow.o mainwindow.cpp
mainwindow.cpp: In member function 'void MainWindow::SaveProject()':
mainwindow.cpp:897:29: error: 'endl' was not declared in this scope; did you mean 'Qt::endl'?
  897 |         str << "PROJECT" << endl;
      |                             ^~~~
      |                             Qt::endl
In file included from ../../../../system/develop/headers/x86/QtCore/QTextStream:1,
                 from mainwindow.cpp:12:
../../../../system/develop/headers/x86/QtCore/qtextstream.h:261:28: note: 'Qt::endl' declared here
  261 | Q_CORE_EXPORT QTextStream &endl(QTextStream &s);
      |                            ^~~~
Makefile:1107: recipe for target '../obj/mainwindow.o' failed
make: *** [../obj/mainwindow.o] Error 1
`

**Comment by plattysoft** [comment]
I was just trying to compile it in Ubuntu with the same result.
I know I have compiled it in Raspbian before, maybe the current version of QT is more strict now, but IIRC I compiled it with QT5 both times.
Even if you do a search / replace (in a lot of places) there is another error later on about a pointer not being a pointer and not able to use ->

I'll let you know if I manage to compile it. I used it a lot on Mac, and I want to use it in Linux as well.

**Comment by plattysoft** [comment]
Ok, lo he compilado en Ubuntu.
Solamente tienes que cambiar todas las ocurrencias de `endl` por `Qt::endl` y en csprites.cpp hay que cambiar `->toImage()` por `.toImage()` y compila sin problemas.
No he probado aun a exportar ficheros, pero deberia funcionar sin problemas.

**Comment by Pentacour** [comment]
Sí, Okk, gracias. Qt6 ha cambiado alguna cosa y no compilaba. He hecho esos cambios y uno más en el tratamiento del click del ratón.

Como pipagerardo ahora no tiene tiempo, retomo yo el desarrollo en el fork de mi cuenta. Si salen errores o propuestas, miraré de añadirlo si no es mucho lío ;-) De momento el código para que compile en Qt6 está ya allí.

**Comment by plattysoft** [comment]
Muchas gracias!

**Comment by unspacyar** [comment]
Hola!  Seguí las instrucciones que amablemente indicó @plattysoft  y pude compilarlo correctamente.  Muchas gracias por la ayuda!!!!

**Comment by rpelorosso-tw** [comment]
Si están usando Archlinux, cree un paquete para el AUR (nmsxtiles-git)
